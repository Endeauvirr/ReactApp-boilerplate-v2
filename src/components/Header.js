import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { startLogout } from '../actions/auth';

export const Header = ({ photoURL, displayName, startLogout }) => (
  <header className="loggedUser__header--wrapper">
    <div className="header--nav-section">
      <h1 className="header--logo">
        <NavLink to="/dashboard">
          Boilerplate App
        </NavLink>
      </h1>
      <ul className="header--navigation">
        <li>
          <NavLink to="/dashboard" activeClassName="is-active" exact={true}>Dashboard</NavLink>
        </li>
      </ul>
    </div>
    <div className="header--user-section">
      <div className="user-section__photo">
        <img src={photoURL} alt={displayName} />
      </div>
      <div className="user-section__name--wrapper">
        <p className="user-section__name--label">Logged as:</p>
        <p className="user-section__name">{displayName}</p>
      </div>
      <div className="user-section__menu">
        <button onClick={startLogout}>Log out</button>
      </div>
    </div>
  </header>
);

const mapStateToProps = (state) => ({
  photoURL: state.auth.photoURL,
  displayName: state.auth.displayName
});

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
