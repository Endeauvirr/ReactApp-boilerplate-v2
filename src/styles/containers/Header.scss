.loggedUser__header--wrapper {
  width: 100%;
  display: flex;
  justify-content: space-between;
  padding: 0 25px;
  border-bottom: 1px solid $color-light-gray;
}

.header--nav-section {
  width: auto;
  display: flex;
  align-items: stretch;
  .header--logo {
    margin: 0 30px 0 0;
    align-self: center;
    a {
      @include primaryTypeface($weight: 700);
      color: $color-rich-black;
      text-decoration: none;
    }
  }
  .header--navigation {
    display: inline-flex;
    list-style: none;
    margin: 0;
    padding: 0;
    border-right: 1px solid $color-light-gray;

    li {
      display: flex;
      align-items: center;
      border-left: 1px solid $color-light-gray;
      background: #fff;
      box-shadow: none;
      transition: all 0.3s linear;

      &:hover {
        box-shadow: 0 5px 10px 0 rgba(0,0,0,0.15);
      }

      a {
        padding: 30px;
      }
    }
  }
}

.header--user-section {
  width: auto;
  display: flex;

  .user-section {
    &__photo {
      width: auto;
      display: flex;
      align-items: center;
      img {
        width: 50px;
        border-radius: 50%;
      }
    }

    &__name--wrapper {
      width: auto;
      display: grid;
      align-self: center;
    }
    &__name--label {
      @include primaryTypeface($weight: 300);
      font-size: 0.875em;
      width: 100%;
      margin: 0 0 5px 0;
    }
    &__name {
      @include primaryTypeface($weight: 700);
      margin: 0;
    }
  }
}