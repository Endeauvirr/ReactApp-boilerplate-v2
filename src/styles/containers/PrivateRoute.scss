.loggedUser--wrapper {
  display: grid;
  grid-template-columns: 100%;
  grid-template-rows: 80px auto;
}