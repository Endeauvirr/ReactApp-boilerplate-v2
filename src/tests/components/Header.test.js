import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../containers/Header';

let startLogout;
let photoURL;
let displayName;

beforeEach(() => {
  startLogout = jest.fn();
  photoURL = 'picture.jpg';
  displayName = 'name surname';
});

test('Should render Header container correctly', () => {
  const wrapper = shallow(<Header startLogout={() => {}} photoURL={photoURL} displayName={displayName} />);
  expect(wrapper).toMatchSnapshot();
});

test('Should call startLogout on button click', () => {
  const startLogout = jest.fn();
  const wrapper = shallow(<Header startLogout={startLogout} />);

  wrapper.find('button').simulate('click');
  expect(startLogout).toHaveBeenCalled();
});
