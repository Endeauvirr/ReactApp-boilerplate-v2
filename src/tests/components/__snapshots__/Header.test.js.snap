// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header container correctly 1`] = `
<header
  className="loggedUser__header--wrapper"
>
  <div
    className="header--nav-section"
  >
    <h1
      className="header--logo"
    >
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        to="/dashboard"
      >
        Boilerplate App
      </NavLink>
    </h1>
    <ul
      className="header--navigation"
    >
      <li>
        <NavLink
          activeClassName="is-active"
          ariaCurrent="true"
          exact={true}
          to="/dashboard"
        >
          Dashboard
        </NavLink>
      </li>
    </ul>
  </div>
  <div
    className="header--user-section"
  >
    <div
      className="user-section__photo"
    >
      <img
        alt="name surname"
        src="picture.jpg"
      />
    </div>
    <div
      className="user-section__name--wrapper"
    >
      <p
        className="user-section__name--label"
      >
        Logged as:
      </p>
      <p
        className="user-section__name"
      >
        name surname
      </p>
    </div>
    <div
      className="user-section__menu"
    >
      <button
        onClick={[Function]}
      >
        Log out
      </button>
    </div>
  </div>
</header>
`;
